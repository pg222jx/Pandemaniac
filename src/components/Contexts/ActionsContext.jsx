import React,{ useState, createContext } from 'react';

export const ActionsContext = createContext()

export const ActionsProvider = props => {

    const [amount, setAmount] = useState(2)

    return (
        <ActionsContext.Provider value={[amount, setAmount]}>
            {props.children}
        </ActionsContext.Provider>
    )
}

