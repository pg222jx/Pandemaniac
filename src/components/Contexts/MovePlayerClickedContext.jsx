import React,{ useState, createContext } from 'react';

export const MovePlayerClickedContext = createContext()

export const MovePlayerClickedProvider = props => {

    const [clicked, setClicked] = useState(false)

    return (
        <MovePlayerClickedContext.Provider value={[clicked, setClicked]}>
            {props.children}
        </MovePlayerClickedContext.Provider>
    );
}

