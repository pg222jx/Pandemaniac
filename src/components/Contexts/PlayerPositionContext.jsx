import React,{ useState, createContext } from 'react';

export const PlayerPositionContext = createContext()

export const PlayerPositionProvider = props => {

    const [position, setPosition] = useState({bottom: undefined, left: undefined})

    return (
        <PlayerPositionContext.Provider value={[position, setPosition]}>
            {props.children}
        </PlayerPositionContext.Provider>
    )
}

