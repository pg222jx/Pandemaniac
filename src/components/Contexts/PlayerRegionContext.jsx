import React,{ useState, createContext } from 'react';

export const PlayerRegionContext = createContext()

export const PlayerRegionProvider = props => {

    const [region, setRegion] = useState('')

    return (
        <PlayerRegionContext.Provider value={[region, setRegion]}>
            {props.children}
        </PlayerRegionContext.Provider>
    )
}

