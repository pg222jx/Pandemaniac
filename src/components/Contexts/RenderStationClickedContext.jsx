import React,{ useState, createContext } from 'react';

export const RenderStationClickedContext = createContext()

export const RenderStationClickedProvider = props => {

    const [clicked, setClicked] = useState(false)

    return (
        <RenderStationClickedContext.Provider value={[clicked, setClicked]}>
            {props.children}
        </RenderStationClickedContext.Provider>
    );
}

