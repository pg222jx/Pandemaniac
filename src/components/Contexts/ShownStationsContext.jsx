import React, { useState, createContext } from 'react'

export const ShownStationsContext = createContext()

export const ShownStationsProvider = props => {

    const [shownStations, setShownStations] = useState([])

    return (
        <ShownStationsContext.Provider value={[shownStations, setShownStations]}>
            {props.children}
        </ShownStationsContext.Provider>
    )
}

