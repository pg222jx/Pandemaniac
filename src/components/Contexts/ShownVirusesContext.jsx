import React, { useState, createContext } from 'react';

export const ShownVirusesContext = createContext()

export const ShownVirusesProvider = props => {

    const [shownViruses, setShownViruses] = useState([])

    return (
        <ShownVirusesContext.Provider value={[shownViruses, setShownViruses]}>
            {props.children}
        </ShownVirusesContext.Provider>
    )
}

