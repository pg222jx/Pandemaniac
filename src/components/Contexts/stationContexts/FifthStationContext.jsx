import React,{ useState, createContext } from 'react';

export const FifthStationContext = createContext()

export const FifthStationProvider = props => {

    const [style, setStyle] = useState({bottom: undefined, left: undefined, visibility: 'hidden'})

    return (
        <FifthStationContext.Provider value={[style, setStyle]}>
            {props.children}
        </FifthStationContext.Provider>
    );
}

