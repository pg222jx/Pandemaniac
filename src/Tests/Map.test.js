import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import { PlayerPositionProvider } from '../components/Contexts/PlayerPositionContext'
import { ActionsProvider } from '../components/Contexts/ActionsContext'
import { FirstStationProvider } from '../components/Contexts/stationContexts/FirstStationContext'
import { SecondStationProvider } from '../components/Contexts/stationContexts/SecondStationContext'
import { ThirdStationProvider } from '../components/Contexts/stationContexts/ThirdStationContext'
import { FourthStationProvider } from '../components/Contexts/stationContexts/FourthStationContext'
import { FifthStationProvider } from '../components/Contexts/stationContexts/FifthStationContext'
import { SixthStationProvider } from '../components/Contexts/stationContexts/SixthStationContext'
import { MovePlayerClickedProvider } from '../components/Contexts/MovePlayerClickedContext'
import { RenderStationClickedProvider } from '../components/Contexts/RenderStationClickedContext'
import { PlayerRegionProvider } from '../components/Contexts/PlayerRegionContext'
import { ShownVirusesProvider } from '../components/Contexts/ShownVirusesContext'
import { ShownStationsProvider } from '../components/Contexts/ShownStationsContext'
import Map from '../components/Map'


const wrapperShallow = shallow(
    <PlayerPositionProvider>
        <ActionsProvider>
            <FirstStationProvider>
                <SecondStationProvider>
                    <ThirdStationProvider>
                        <FourthStationProvider>
                            <FifthStationProvider>
                                <SixthStationProvider>
                                    <MovePlayerClickedProvider>
                                        <RenderStationClickedProvider>
                                            <PlayerRegionProvider>
                                                <ShownVirusesProvider>
                                                    <ShownStationsProvider>
                                                        <Map/>
                                                    </ShownStationsProvider>
                                                </ShownVirusesProvider>
                                            </PlayerRegionProvider>
                                        </RenderStationClickedProvider>
                                    </MovePlayerClickedProvider>
                                </SixthStationProvider>
                            </FifthStationProvider>
                        </FourthStationProvider>
                    </ThirdStationProvider>
                </SecondStationProvider>
            </FirstStationProvider>
        </ActionsProvider>
    </PlayerPositionProvider>
)


describe('testing the Map component', () => {
    it('should match the snapshot', () => {
        expect(wrapperShallow.html()).toMatchSnapshot()
      })

      it('should render without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(
        <PlayerPositionProvider>
            <ActionsProvider>
                <FirstStationProvider>
                    <SecondStationProvider>
                        <ThirdStationProvider>
                            <FourthStationProvider>
                                <FifthStationProvider>
                                    <SixthStationProvider>
                                        <MovePlayerClickedProvider>
                                            <RenderStationClickedProvider>
                                                <PlayerRegionProvider>
                                                    <ShownVirusesProvider>
                                                        <ShownStationsProvider>
                                                            <Map/>
                                                        </ShownStationsProvider>
                                                    </ShownVirusesProvider>
                                                </PlayerRegionProvider>
                                            </RenderStationClickedProvider>
                                        </MovePlayerClickedProvider>
                                    </SixthStationProvider>
                                </FifthStationProvider>
                            </FourthStationProvider>
                        </ThirdStationProvider>
                    </SecondStationProvider>
                </FirstStationProvider>
            </ActionsProvider>
        </PlayerPositionProvider>, div
        )
       })
})
