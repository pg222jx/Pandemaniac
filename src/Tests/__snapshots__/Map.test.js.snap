// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the Map component should match the snapshot 1`] = `"<img src=\\"map.png\\" alt=\\"World map\\" class=\\"worldMap\\" useMap=\\"#mapImage\\"/><map name=\\"mapImage\\"><area href=\\"#\\" alt=\\"North America\\" shape=\\"rect\\" coords=\\"20,20, 400,240\\"/><area href=\\"#\\" alt=\\"South America\\" shape=\\"rect\\" coords=\\"155,255, 350,520\\"/><area href=\\"#\\" alt=\\"Europe\\" shape=\\"rect\\" coords=\\"435,20, 630,140\\"/><area href=\\"#\\" alt=\\"Asia\\" shape=\\"rect\\" coords=\\"640,20, 950,200\\"/><area href=\\"#\\" alt=\\"Africa\\" shape=\\"rect\\" coords=\\"435,170, 580,450\\"/><area href=\\"#\\" alt=\\"Australia\\" shape=\\"rect\\" coords=\\"800,300, 1100,480\\"/></map>"`;
